shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -257,6 +253,23 @@
 }
 
 void FlutterPlatformViewsController::DetachUnusedLayers() {
+  // Clean up disposed views.
```

Comment:
It's possible that the composition order didn't change, in which case we will return on line 229 and won't dispose the views that should be disposed.

I'd suggest splitting the part that disposes views out of this method, and call `DisposeViews` earlier in `SubmitFrame`, as this method currently is doing more than what its name suggests.

Reply:
Done